src/wire/ip.rs

```diff
@@ -336,6 +337,26 @@ impl IpRepr {
         }
     }
 
+    /// Return the TTL value.
+    pub fn ttl(&self) -> u8 {
+        match self {
+            &IpRepr::Unspecified { ttl, .. } => ttl,
+            &IpRepr::Ipv4(Ipv4Repr { ttl, .. }) => ttl,
+            &IpRepr::__Nonexhaustive => unreachable!()
+        }
+    }
+
+    /// Set the TTL value.
+    pub fn set_ttl(&mut self, limit: u8) {
```

Comment:
This isn't super useful. I'd be okay with backing this out.

Reply:
Let's leave this out, `set_payload_len` is special here.